grpc/creds/creds.go

```diff
@@ -13,28 +12,41 @@ import (
 // transportCredentials is a grpc/credentials.TransportCredentials which supports
 // connecting to, and verifying multiple DNS names
 type transportCredentials struct {
-	roots   *x509.CertPool
-	clients []tls.Certificate
+	clientConfig *tls.Config
```

Comment:
It seems like there is no advantage to having a single object that can contain either a clientConfig or serverConfig. As far as I can tell, the two types don't share much in common. Wouldn't it make more sense to define a `clientTransportCredentials` that always returned error from `ServerHandshake`, and a `serverTransportCredentials` that always returned error from `ClientHandshake`?

Also, I'm not sure about the utility of switching from `rootCAs` and `clientCerts` to a `tls.Config`. I think using the `tls.Config` as a container makes it a little less clear what is being copied and what is being overridden. What was your goal with that change?